pkgs/development/python-modules/jaxlib/default.nix

```diff
@@ -232,17 +239,31 @@ let
       # 4) Patch tensorflow sources to work with later versions of protobuf. See
       #    https://github.com/google/jax/issues/9534. Note that this should be
       #    removed on the next release after 0.3.0.
+      # 5) Patch in https://github.com/bazelbuild/rules_cc/pull/136 to fix the bazel compiler detection on aarch64-darwin
       preBuild = ''
+        patch -p 0 -d ../output/external/rules_cc/ -i "${./macos-arm64.patch}"
         for src in ./jaxlib/*.{cc,h}; do
           sed -i 's@include/pybind11@pybind11@g' $src
         done
-        sed -i 's@-lprotobuf@-l:libprotobuf.a@' ../output/external/org_tensorflow/third_party/systemlibs/protobuf.BUILD
-        sed -i 's@-lprotoc@-l:libprotoc.a@' ../output/external/org_tensorflow/third_party/systemlibs/protobuf.BUILD
         substituteInPlace ../output/external/org_tensorflow/tensorflow/compiler/xla/python/pprof_profile_builder.cc \
           --replace "status.message()" "std::string{status.message()}"
+        substituteInPlace ../output/external/rules_cc/cc/private/toolchain/osx_cc_wrapper.sh.tpl \
+          --replace "/usr/bin/install_name_tool" "${cctools}/bin/install_name_tool"
+        substituteInPlace ../output/external/rules_cc/cc/private/toolchain/unix_cc_configure.bzl \
+          --replace "/usr/bin/libtool" "${cctools}/bin/libtool"
```

Comment:
I put some of this patching that is compatible with all platforms in the common block, so the patching can be easily tested on all platforms when updating.